File: unity/ProfitableCore/Schema/InitiateVoyageResult.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/initiateVoyageResult.ts. Fuel deduction is a real
// Ship state change the caller must persist -- same "return what
// changed" discipline every other core function in this codebase follows.
public class InitiateVoyageResult
{
    public Voyage Voyage { get; set; } = new();
    public Ship UpdatedShip { get; set; } = new();
}

File: src/data/types/initiateVoyageResult.ts
import type { Voyage } from "./voyage.ts";
import type { Ship } from "./ship.ts";

// Ship Fuel amendment: initiateVoyage() used to return a bare Voyage: fuel
// deduction is a real Ship state change the caller must persist, so the
// return shape grew to include it -- same "return what changed"
// discipline every other core function in this codebase follows.
export interface InitiateVoyageResult {
  voyage: Voyage;
  updatedShip: Ship;
}
